implement getContactsByPhoneNumber [DONE]

hatemalimam opened the pull request:
as discussed here https://github.com/rt2zz/react-native-contacts/issues/137#issuecomment-515280224
I'm submitting the PR that implements  `getContactsByPhoneNumber` on both platforms.

the changes are tested and can be tested on this branch https://github.com/hatemalimam/react-native-contacts-test/tree/feature/getContactsByPhoneNumber

morenoh149 commented:
Awesome work. The test repo has actually been deprecated (I'll archive it now). Instead demonstrate how to use this new api in the example https://github.com/rt2zz/react-native-contacts/blob/master/example/App.js

hatemalimam commented:
I have updated the example app.
I hope you don't mind, I have taken the liberty to revamp the example app a bit, since it was very basic.

It includes the basic functionalities of this lib.

I have updated also the README accordingly.

The app is tested against iOS and android.

![react-native-contacts](https://user-images.githubusercontent.com/3439970/61995542-8e434d80-b092-11e9-97e2-7b77a8be568c.gif)

morenoh149 commented on android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java:
```diff
@@ -138,6 +138,38 @@ protected Void doInBackground(final Void ... params) {
+     * @param callback user provided callback to run at completion
+     */
+    private void getAllContactsByPhoneNumber(final String phoneNumber, final Callback callback) {
+        AsyncTask<Void,Void,Void> myAsyncTask = new AsyncTask<Void,Void,Void>() {
```
This function body should be inlined into `getContactsByPhoneNumber` on line 147. I think `getAllContactsMatchingString` was split into two function due to misunderstanding what was going on in `getAll`,`getAllWithoutPhotos` and `getAllContacts`.

@hatemalimam is it necessary for `getAllContactsByPhoneNumber` to be two functions? one decorated by `@ReactMethod` and one not decorated.

hatemalimam replied:
In fact I see no reason to have them split like that, I was just following the one done in `getAllContactsMatchingString`, as it implies some convention.

I'll submit a refactor PR.

morenoh149 replied:
awesome! Things like this happen when multiple people work on a codebase, you don't remember why something was done the way it was. And the comments don't explain either.

morenoh149 commented:
thanks!

morenoh149 commented:
I tried using the readme images but couldn't get them to work. If you want to resubmit them feel free.

jamesawer3 commented:
Is getContactsByPhoneNumber working in iOS ??? For me its not working updated react-native-contacts to latest version and tried but its saying getContactsByPhoneNumber not defined.

hatemalimam commented:
I guess it's not released yet, not sure why, the current version is `5.0.2`, it's supposed to be released with `5.0.3`. however you can always add the dependency from the master branch if it's needed critically. 

@morenoh149 any plans on a soon release ?

morenoh149 commented:
I cut a release now.

prozwang commented:
awesome ! cool ! Yet, there still one place need to be optimized
search for phone number is not good to use
if i want to find a friend in my contacts book, but i only remember part of his phone number 
i can't find him except i input his whole phone number

e.g.   Name: Tom.  PhoneNumber: 888812345
only input '888812345' ,i can find tom
but i expect when i input '888' or '12345' , my search list would show tom

hatemalimam commented:
Well in android it would work as you've expected, but the provided api at the iOS side has an exact match, maybe I'll look into it later to see if we can support the same in iOS.
